services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVNCHandler.java

```diff
@@ -104,20 +105,20 @@ public void onConnect(final Session session) throws IOException, InterruptedExce
         try {
             port = Integer.parseInt(portStr);
         } catch (NumberFormatException e) {
-            logger.error("Invalid port value in query string: {}. Expected a number.", portStr, e);
+            logger.error(String.format("Invalid port value in query string: %s. Expected a number.", portStr), e);
```

Comment:
can you undo this change?

Reply:
I'm not really sure why, but the `logger` fields of `com.cloud.consoleproxy.ConsoleProxyNoVNCHandler` and `com.cloud.consoleproxy.vnc.NoVncClient` are not instantiated directly from `log4j`. They're generated from a static method from the `com.cloud.consoleproxy.util.Logger` helper class.

The problem is that this helper class does not implement the new APIs provided by the new version of `log4j`, especially those that accept multiple parameters. As a consequence of that, `String.format()` must be used to log messages in those two classes.

IMO, it should be evaluated if the helper class is still needed. If so, we need to implement new methods to be able to use the new syntax provided by `log4j2`.

Reply:
I see, That was a good initiative back in 2013, to create a library facade, it had not been implemented fully in the project though;
```
// logger facility for dynamic switch between console logger used in Applet and log4j based logger
```
This might be an oversight in the log4j migration. `com.cloud.consoleproxy.ConsoleProxyLoggerFactory` is still using log4j v1 classes under the hood.

I personally think we should have implemented such a proxy library, for the whole project and implemented log4j under the hood. I think for now we will have to do with 'fixing' the console proxy classes.